source/darwin/darwin_pki_utils.c

```diff
@@ -105,9 +105,40 @@ int aws_import_public_and_private_keys_to_identity(
     }
 
     if (key_status != errSecSuccess && key_status != errSecDuplicateItem) {
-        AWS_LOGF_ERROR(AWS_LS_IO_PKI, "static: error importing private key with OSStatus %d", (int)key_status);
-        result = aws_raise_error(AWS_IO_FILE_VALIDATION_FAILURE);
-        goto done;
+        if (key_status == errSecUnknownFormat) { // If the format is unknow, try ecc
```

Comment:
if nesting seems to be off here. you can probably pull the "if (key_status == errSecUnknownFormat)" up and then "if (key_status != errSecSuccess && key_status != errSecDuplicateItem)" can be checked once